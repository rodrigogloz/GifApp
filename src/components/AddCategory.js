import React, { useState } from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrashCan, faSearch } from "@fortawesome/free-solid-svg-icons";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 0) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  const resetCategories = () => {
    setCategories([]);
  };

  return (
    <div className="search-wrapper">
      <form onSubmit={handleSubmit} className="search-container">
        <input
          spellCheck="false"
          className="animate__animated animate__fadeIn"
          type="text"
          placeholder="Buscar GIFs por categorÃ­a"
          value={inputValue}
          onChange={handleInputChange}
        />
        <button className="search-button" onClick={handleSubmit}>
          <FontAwesomeIcon icon={faSearch}></FontAwesomeIcon>
        </button>
      </form>

      <select className="search-quantity">
        <option value="1">10</option>
        <option value="2">50</option>
        <option value="3">100</option>
      </select>

      <button className="reset-button" onClick={resetCategories}>
        <FontAwesomeIcon icon={faTrashCan}></FontAwesomeIcon>
      </button>
    </div>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
