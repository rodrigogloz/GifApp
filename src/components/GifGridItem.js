import React from "react";
import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
  return <img src={url} alt={title} title={title} className="card card-img" />;
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
