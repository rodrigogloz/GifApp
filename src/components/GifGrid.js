import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import deleteButton from "../assets/delete-svgrepo-com.svg";
// import loadingWheel from "../assets/Spin-1s-200px.gif";

export const GifGrid = ({ category }) => {
  const { state, setState } = useFetchGifs(category);
  const { data: images, loading } = state;

  const onDelete = () => {
    setState({
      data: [],
      loading: true,
    });
  };

  return (
    images.length > 0 && (
      <div className="category-container">
        <span className="category-header animate__animated animate__fadeIn">
          <h3 className="category-name animate__animated animate__fadeIn">
            {category}
          </h3>
          <img
            src={deleteButton}
            alt="delete"
            className="icon-delete"
            onClick={onDelete}
          />
        </span>
        {loading && <p>Cargando...</p>}

        <div className="card-grid scrolling-wrapper">
          {images.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </div>
      </div>
    )
  );
};
