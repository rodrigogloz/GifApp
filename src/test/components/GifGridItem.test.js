import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas con el componente <GifgridItem/>", () => {
  const title = "Un título";
  const url = "http://localhost/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  //   test("Debe de tener un párrafo con el title", () => {
  //     const p = wrapper.find("p");
  //     expect(p.text().trim()).toBe(title);
  //   });

  //   test("Debe tener la imagen igual al url y alt de los props", () => {
  //     const img = wrapper.find("img");

  //     expect(img.props("src")).toBe(url);
  //   });

  test("Debe tener el animate__fadeIn", () => {
    const div = wrapper.find("div");

    const className = div.prop("className");

    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
