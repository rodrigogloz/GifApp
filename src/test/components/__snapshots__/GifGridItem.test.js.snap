// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente <GifgridItem/> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="http://localhost/algo.jpg"
    title="Un título"
  />
</div>
`;
